src/material/badge/badge.spec.ts

```diff
@@ -330,3 +381,15 @@ class NestedBadge {}
     <ng-template matBadge="1">Notifications</ng-template>`,
 })
 class BadgeOnTemplate {}
+
+@Component({template: `<mat-icon matBadge="1">home</mat-icon>`})
+class IconWithBadge {}
```

Comment:
We should follow the naming convention used on the previous test components. E.g.
`BadgeOnIcon`